src/apify_helpers.js

```diff
@@ -129,10 +130,34 @@ const enrichActorRun = async (z, run, storeKeysToInclude = []) => {
     return _.omit(run, OMIT_ACTOR_RUN_FIELDS);
 };
 
+// Get actor statuses from bundle input data. This method makes sure that each status is unique and that it is valid
+const getActorStatusesFromBundle = (bundle) => {
+    if (bundle.inputData.statuses && bundle.inputData.statuses.length > 0) {
+        const statuses = Array.from(new Set(bundle.inputData.statuses));
+        const filteredStatuses = statuses
+            .filter((status) => Object.keys(ACTOR_RUN_TERMINAL_STATUSES).includes(status));
+
+        return filteredStatuses.length > 0 ? filteredStatuses : ACTOR_JOB_TERMINAL_STATUSES;
+    }
+
+    return ACTOR_JOB_TERMINAL_STATUSES;
+};
+
 // Process to subscribe to Apify webhook
 const subscribeWebhook = async (z, bundle, condition) => {
+    const statuses = getActorStatusesFromBundle(bundle);
+
+    const eventTypes = statuses.map((status) => status.replace('-', '_')).map((status) => {
+        // eslint-disable-next-line no-restricted-syntax
+        for (const eventType of Object.values(WEBHOOK_EVENT_TYPES)) {
```

Comment:
This is not nice you need to loop every time.
What about doing a simple key-value object or Map where you map status to event time and use this?
You would not need to replace '-', '_'.